central/complianceoperator/v2/report/manager/watcher/scanwatcher.go

```
@@ -0,0 +1,239 @@
+)
+
+// ScanWatcher determines if a scan is running or has completed.
+type ScanWatcher interface {
+	PushScan(v2 *storage.ComplianceOperatorScanV2) error
+	PushCheckResult(*storage.ComplianceOperatorCheckResultV2) error
+	Finished() concurrency.ReadOnlySignal
+	Stop()
+}
+
+// ScanWatcherResults is returned when the watcher detects that the scan is completed.
+type ScanWatcherResults struct {
+	Ctx          context.Context
+	WatcherID    string
+	Scan         *storage.ComplianceOperatorScanV2
+	CheckResults set.StringSet
+	Error        error
+}
+
+// IsComplianceOperatorHealthy indicates whether Compliance Operator is ready for automatic reporting
+func IsComplianceOperatorHealthy(ctx context.Context, clusterID string, complianceIntegrationDataStore complianceIntegrationDS.DataStore) error {
+	coStatus, err := complianceIntegrationDataStore.GetComplianceIntegrationByCluster(ctx, clusterID)
+	if err != nil {
+		return errors.Wrap(err, ComplianceOperatorIntegrationDataStoreError.Error())
+	}
+	if len(coStatus) == 0 {
+		log.Errorf("No compliance integrations retrieved from cluster %s", clusterID)
+		return ComplianceOperatorIntegrationZeroIntegrationsError
+	}
+	if !coStatus[0].GetOperatorInstalled() {
+		return ComplianceOperatorNotInstalledError
+	}
+	if semver.Compare(coStatus[0].GetVersion(), minimumComplianceOperatorVersion) < 0 {
+		return ComplianceOperatorVersionError
+	}
+	return nil
+}
+
+// GetWatcherIDFromScan given a Scan, returns a unique ID for the watcher
+func GetWatcherIDFromScan(scan *storage.ComplianceOperatorScanV2) (string, error) {
+	clusterID := scan.GetClusterId()
+	if clusterID == "" {
+		return "", errors.New("Missing cluster ID")
+	}
+	scanID := scan.GetId()
+	if scanID == "" {
+		return "", errors.New("Missing scan ID")
+	}
+	startTime := scan.GetLastStartedTime()
+	if startTime == nil {
+		// This could happen if the scan is freshly created
+		log.Debug("Missing last stated time")
+		return "", nil
```

What happens here?  Does the caller handle the empty ID return?  Should we return an err instead?